### ISSUE TITLE
Parser delivers illegal values for notes when they are not separated by spaces

### ISSUE THREAD

**Issue by ftolsson** [lmparppei/Beat#128]
Small one. Fixed it on my side already for FTOutliner, but wanted to let you know FYI:

If a user adds two color tags to the document (by forcefully typing them in) like so:
```
INT. SOMEWHERE -- DAY [[COLOR RED]][[COLOR BLUE]]
```
the API will give the scene/section/synopsis the following value:
```
{...
"color" : "RED]][[COLOR BLUE",
...}
```

In fact, adding any note after a ```[[COLOR SOMETHING]]``` tag (unless separating them with spaces) will parse the entire thing as a color value for purpose of the API.

**Comment by lmparppei** [comment]
This is caused by parser's reliance on `NSIndexSet`s. When enumerating indices as ranges, continuous ranges are enumerated as a single one. I'll have to refactor `colorForHeading:`.

**Comment by lmparppei** [comment]
Fixed in `1.98.1`

**Comment by lmparppei** [comment]
On second thought, I'll reopen this because the same change has to be implemented for every single note-based element.
